packages/vitest/src/runtime/runners/benchmark.ts

```diff
@@ -90,7 +94,12 @@ async function runBenchmarkSuite(suite: Suite, runner: NodeBenchmarkRunner) {
       await task.warmup()
       tasks.push([
         await new Promise<BenchTask>(resolve => setTimeout(async () => {
-          resolve(await task.run())
+          let beforeEachCleanups: HookCleanupCallback[] = []
+          beforeEachCleanups = await callSuiteHook(benchmark.suite, benchmark, 'beforeEach', runner, [benchmark.context, benchmark.suite])
```

Comment:
Can we use tinybench's hooks? I don't have any issues with this, but just asking.

Reply:
Interesting idea!
I think we might need to refine what each hook does, since in benchmarking there is one more level of hooks.

At the moment, with my implementation, this is what would happen (I added some comment at the end to see the execution order):

```typescript
import {
  afterAll,
  afterEach,
  beforeAll,
  beforeEach,
  bench,
  describe,
} from 'vitest'

describe('hooks', () => {
  beforeAll(() => {
    console.log('beforeAll')
  })
  beforeEach(() => {
    console.log('beforeEach')
  })
  afterEach(() => {
    console.log('afterEach')
  })
  afterAll(() => {
    console.log('beforeAll')
  })

  bench('one', () => {
    console.log('bench one')
  }, {
    setup: () => {
      console.log('setup')
    },
    teardown: () => {
      console.log('tearDown')
    },
    iterations: 10,
    time: 1,
  })
})

// when the bench is executed, the following should be logged

// beforeAll
// beforeEach
// setup     x10
// bench one x10
// tearDown  x10
// afterEach
// beforeAll
```

Do you think it is what should be expected?

As far as using the hooks from tinybench, I can map the `*Each` hooks from vitest to the `Task` `*All` hooks from tinybench.

Reply:
Oh interesting edge case, nice, I think let's keep it as is and this should be documented somewhere. 

After all, @sheremet-va might have some interesting ideas and insights around this.

Reply:
Okay, I will go and try to add a test that verify the order of execution of the hooks

Reply:
I will switch the PR to draft for the moment, and mark it as ready when I will have added the tests